x-pack/plugins/lens/public/app_plugin/lens_top_nav.tsx

```diff
@@ -164,124 +164,163 @@ export const LensTopNavMenu = ({
   const unsavedTitle = i18n.translate('xpack.lens.app.unsavedFilename', {
     defaultMessage: 'unsaved',
   });
-  const topNavConfig = getLensTopNavConfig({
-    showSaveAndReturn: Boolean(
-      isLinkedToOriginatingApp &&
-        // Temporarily required until the 'by value' paradigm is default.
-        (dashboardFeatureFlag.allowByValueEmbeddables || Boolean(initialInput))
-    ),
-    enableExportToCSV: Boolean(isSaveable && activeData && Object.keys(activeData).length),
-    isByValueMode: getIsByValueMode(),
-    allowByValue: dashboardFeatureFlag.allowByValueEmbeddables,
-    showCancel: Boolean(isLinkedToOriginatingApp),
-    savingToLibraryPermitted,
-    savingToDashboardPermitted,
-    actions: {
-      exportToCSV: () => {
-        if (!activeData) {
-          return;
-        }
-        const datatables = Object.values(activeData);
-        const content = datatables.reduce<Record<string, { content: string; type: string }>>(
-          (memo, datatable, i) => {
-            // skip empty datatables
-            if (datatable) {
-              const postFix = datatables.length > 1 ? `-${i + 1}` : '';
+  const topNavConfig = useMemo(
+    () =>
+      getLensTopNavConfig({
+        showSaveAndReturn: Boolean(
+          isLinkedToOriginatingApp &&
+            // Temporarily required until the 'by value' paradigm is default.
+            (dashboardFeatureFlag.allowByValueEmbeddables || Boolean(initialInput))
+        ),
+        enableExportToCSV: Boolean(isSaveable && activeData && Object.keys(activeData).length),
+        isByValueMode: getIsByValueMode(),
+        allowByValue: dashboardFeatureFlag.allowByValueEmbeddables,
+        showCancel: Boolean(isLinkedToOriginatingApp),
+        savingToLibraryPermitted,
+        savingToDashboardPermitted,
+        actions: {
+          exportToCSV: () => {
+            if (!activeData) {
+              return;
+            }
+            const datatables = Object.values(activeData);
+            const content = datatables.reduce<Record<string, { content: string; type: string }>>(
+              (memo, datatable, i) => {
+                // skip empty datatables
+                if (datatable) {
+                  const postFix = datatables.length > 1 ? `-${i + 1}` : '';
 
-              memo[`${lastKnownDoc?.title || unsavedTitle}${postFix}.csv`] = {
-                content: exporters.datatableToCSV(datatable, {
-                  csvSeparator: uiSettings.get('csv:separator', ','),
-                  quoteValues: uiSettings.get('csv:quoteValues', true),
-                  formatFactory: data.fieldFormats.deserialize,
-                }),
-                type: exporters.CSV_MIME_TYPE,
-              };
+                  memo[`${lastKnownDoc?.title || unsavedTitle}${postFix}.csv`] = {
+                    content: exporters.datatableToCSV(datatable, {
+                      csvSeparator: uiSettings.get('csv:separator, ,'),
```

Comment:
This change doesn't look right, did it happen by accident?

Reply:
I think prettier just went wild here. I'll restore it